samples/kvssink_gstreamer_sample.cpp

```diff
@@ -683,28 +693,34 @@ int gstreamer_init(int argc, char *argv[], CustomData *data) {
     /* init GStreamer */
     gst_init(&argc, &argv);
 
-    GstElement *pipeline;
+    GstElement *pipeline, *kvssink;
     int ret;
     GstStateChangeReturn gst_ret;
 
     // Reset first frame pts
     data->first_pts = GST_CLOCK_TIME_NONE;
 
+    kvssink = gst_element_factory_make("kvssink", "kvssink");
```

Comment:
Because it is common to all cases and used in a common section later
`data->fragment_metadata_timer_id = g_timeout_add_seconds(2, reinterpret_cast<GSourceFunc>(put_metadata), kvssink);`